generate.ts

```diff
@@ -42,8 +41,13 @@ var sdkGeneratorGlobals: SdkGenGlobals = {
     // Internal note: We lowercase the argsByName-keys, targetNames, buildIdentifier, and the flags.  Case is maintained for all other argsByName-values, and targets
     argsByName: {}, // Command line args compiled into KVP's
     errorMessages: [], // String list of errors during parsing and loading steps
-    targetOutputPathList: [], // A list of objects that describe sdk targets to build
-    buildFlags: [], // The sdkBuildFlags which modify the list of APIs available in this build of the SDK
+    buildTarget: {
```

Comment:
>buildTarget [](start = 4, length = 11)

This can be simply of a type IBuildTarget, right? (instead of repeating the same field declarations that you already declared in that interface)

Reply:
No, because I'm using interfaces, not objects.
There's a long answer, but the short answer is JS and TS are weird.

Reply:
I meant, you could type it as:

buildTarget: IBuildTarget {
 buildFlags: [] ...
 . . .
}
for additional type safety? no?

---
In reply to: [290496574](https://github.com/PlayFab/SDKGenerator/pull/544#discussion_r290496574) [](ancestors = 290496574)

Reply:
No, the type safety is already strictly enforced, and can't be violated. Let's talk in person if you need.